src/elements/trade/TradeWidgetInput.tsx

```diff
@@ -0,0 +1,200 @@
+import { useAppSelector } from 'store/index';
+import { useTokenInputV3Return } from 'elements/trade/useTknFiatInput';
+import { Image } from 'components/image/Image';
+import { prettifyNumber, toBigNumber } from 'utils/helperFunctions';
+import { useMemo, useRef, useState } from 'react';
+import { Token } from 'services/observables/tokens';
+import { SearchableTokenList } from 'components/searchableTokenList/SearchableTokenList';
+import { ReactComponent as IconChevron } from 'assets/icons/chevronDown.svg';
+import { classNameGenerator } from 'utils/pureFunctions';
+
+interface Props {
+  input?: useTokenInputV3Return;
+  isLoading?: boolean;
+  onFocus?: () => void;
+  label?: string;
+  tokens: Token[];
+  onTokenSelect: (token: Token) => void;
+  disabled?: boolean;
+  errorMsg?: string;
+  excludedTokens?: string[];
+  includedTokens?: string[];
+  disableSelection?: boolean;
+}
+
+export const TradeWidgetInput = ({
+  input,
+  isLoading,
+  onFocus,
+  label,
+  tokens,
+  onTokenSelect,
+  disabled,
+  errorMsg,
+  excludedTokens,
+  includedTokens,
+  disableSelection,
+}: Props) => {
+  const isFiat = useAppSelector((state) => state.user.usdToggle);
+  const inputRef = useRef<HTMLInputElement>(null);
```

Comment:
are you trying to achieve something like this? https://reactjs.org/docs/hooks-reference.html#useimperativehandle